src/configs/substrateProviders.ts

```diff
@@ -37,40 +35,35 @@ interface ChainConfigs {
   [chain: string]: ChainConfig;
 }
 
-// create a custom hasher (512 bits, combo of blake2 and keccak)
-function hasherH512(data: any) {
-  return u8aConcat(blake2AsU8a(data), keccakAsU8a(data));
-}
+const getChainNames = () => {
+  return [checkEnvVariable('REACT_APP_CHAIN_1'), checkEnvVariable('REACT_APP_CHAIN_2')];
+};
 
-export const CHAIN_1 = process.env.CHAIN_1 || 'Rialto';
-export const CHAIN_2 = process.env.CHAIN_2 || 'Millau';
+export const [CHAIN_1, CHAIN_2] = getChainNames();
 
-export const RIALTO_SUBSTRATE_PROVIDER =
-  process.env.REACT_APP_RIALTO_SUBSTRATE_PROVIDER || 'wss://wss.rialto.brucke.link';
-export const MILLAU_SUBSTRATE_PROVIDER =
-  process.env.REACT_APP_MILLAU_SUBSTRATE_PROVIDER || 'wss://wss.millau.brucke.link';
+const getProvider = (provider: string) => new WsProvider(provider);
 
-export const getProvider = (provider: string) => new WsProvider(provider);
+const getTypeAndHasher = (chainNumber: string) =>
+  getCustomTypesAndHasher(checkEnvVariable(`REACT_APP_CHAIN_${chainNumber}`));
 
-export const getSS58Format = (envPrefix: string | undefined, defaultValue: number) => {
-  if (envPrefix) {
-    return parseInt(envPrefix);
-  }
-  return defaultValue;
+const createConfigObject = (chainNumber: string) => {
+  const { types, hasher } = getTypeAndHasher(chainNumber);
+  return {
+    SS58Format: parseInt(checkEnvVariable(`REACT_APP_SS58_PREFIX_CHAIN_${chainNumber}`)),
+    bridgeId: checkEnvVariable(`REACT_APP_BRIDGE_ID_CHAIN_${chainNumber}`),
+    hasher,
+    provider: getProvider(checkEnvVariable(`REACT_APP_SUBSTRATE_PROVIDER_CHAIN_${chainNumber}`)),
+    types
+  };
 };
 
-export const chainsConfigs: ChainConfigs = {
-  Millau: {
-    SS58Format: getSS58Format(process.env.REACT_APP_MILLAU_SS58_PREFIX, 60),
-    bridgeId: process.env.REACT_APP_MILLAU_BRIDGE_ID || 'mlau',
-    hasher: hasherH512,
-    provider: getProvider(MILLAU_SUBSTRATE_PROVIDER),
-    types: customTypesMillau
-  },
-  Rialto: {
-    SS58Format: getSS58Format(process.env.REACT_APP_RIALTO_SS58_PREFIX, 48),
-    bridgeId: process.env.REACT_APP_RIALTO_BRIDGE_ID || 'rlto',
-    provider: getProvider(RIALTO_SUBSTRATE_PROVIDER),
-    types: customTypesRialto
-  }
+export const getChainConfigs = (): ChainConfigs => {
+  checkExpectedVariables();
```

Comment:
Can't we call this function even before we start `React`? IMHO it's better to fail as soon as possible and not wait for some components to render.

Reply:
yes definitely ... i cannot agree more on this